eng/pipelines/release-cli.yml

```diff
@@ -338,33 +326,109 @@ stages:
 
         - pwsh: |
             Expand-Archive -Path mac/azd-darwin-amd64.zip -DestinationPath mac/
             Remove-Item mac/azd-darwin-amd64.zip
           displayName: Extract azd-darwin-amd64 from zip and remove zip
 
         - pwsh: |
-            New-Item -ItemType Directory -Path signed
-            Copy-Item win/ signed/ -Recurse
-            Copy-Item mac/ signed/ -Recurse
+            New-Item -ItemType Directory -Path signed-mac
+            Copy-Item mac/ signed-mac/ -Recurse
           displayName: Copy signing outputs
           condition: always()
 
-        - publish: signed/
-          artifact: signed
+        # TODO: Replace with https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/steps/publish-artifact.yml
+        # when the common engsys is imported.
+        # https://github.com/Azure/azure-dev/issues/956
+        - task: PublishPipelineArtifact@1
           condition: succeeded()
+          displayName: Publish Signed Artifacts
+          inputs:
+            artifactName: signed-mac
+            path: signed-mac/
+
+        - task: PublishPipelineArtifact@1
+          condition: failed()
+          displayName: Publish failed Signed Artifacts
+          inputs:
+            artifactName: signed-mac-FailedAttempt$(System.JobAttempt)
+            path: signed-mac/
+
+    - job: SignWindows
+      pool:
+        name: azsdk-pool-mms-win-2019-general
+        vmImage: MMS2019
+
+      steps:
+        # Checkout required to build MSI
+        - checkout: self
+
+        - task: DownloadPipelineArtifact@2
+          inputs:
+            artifact: azd-windows-amd64.exe
+            path: win
+
+        - task: DownloadPipelineArtifact@2
+          inputs:
+            artifact: install-pwsh
+            path: installer
+
+        - pwsh: Copy-Item installer/*.ps1 win
+          displayName: Copy install scripts to win/
+
+        # TODO: Uncomment before merge
+        # - ${{ if in(variables['Build.Reason'], 'IndividualCI', 'BatchedCI', 'Manual') }}:
+        # TODO: Update ref to appropriate tag before merging
+        - template: pipelines/steps/azd-cli-win-signing.yml@djurek/azd-msi-signing
+          parameters:
+            WinPath:  win
+            WinPattern: '**'
+
+        # TODO: Uncomment before merge
+        # - ${{ else }}:
+        #   - pwsh: Write-Host "Skipping signing. Build reason - $(Build.Reason)"
+        #     displayName: Signing process skipped for non-release build
+
+        - pwsh: |
+            New-Item -ItemType Directory -Path signed-win
+            Copy-Item win/ signed-win/ -Recurse
+            Copy-Item win/azd-windows-amd64.exe cli/azd/azd.exe
+          displayName: Copy signing outputs for publishing and MSI build
+          condition: always()
+
+        - task: MSBuild@1
+          displayName: Build release MSI
+          inputs:
+            solution: cli/installer/windows/azd.wixproj
+            msbuildArguments: /p:RunWixToolsOutOfProc=true
```

Comment:
/p:Configuration=Release

MSIs build Debug by default. Unless Configuration is being set as an env var somewhere prior.